src/tess_asteroids/movingtpf.py

```diff
@@ -669,25 +734,72 @@ def _save_tpf(self, file_name: Optional[str] = None, save_loc: str = ""):
                 ]
             ),
         )
-        table_hdu.header["EXTNAME"] = "PIXELS"
+        table_hdu_spoc.header["EXTNAME"] = "PIXELS"
 
-        # Create aperture HDU
-        # >>>>> UPDATE IN FUTURE WITH REAL APERTURE <<<<<
-        aperture_hdu = fits.ImageHDU(
-            data=np.ones(self.shape),
+        # Create HDU containing average aperture
+        # Aperture has values 0 and 2, where 0/2 indicates the pixel is outside/inside the aperture.
+        # This format is used to be consistent with the aperture HDU from SPOC.
+        aperture_hdu_average = fits.ImageHDU(
+            data=np.nanmedian(self.aperture_mask, axis=0).astype("int32") * 2,
             header=fits.Header(
                 [*self.cube.output_secondary_header.cards, *wcs_header.cards]
             ),
         )
-        aperture_hdu.header["EXTNAME"] = "APERTURE"
-        aperture_hdu.header.set("NPIXSAP", None, "Number of pixels in optimal aperture")
-        aperture_hdu.header.set(
+        aperture_hdu_average.header["EXTNAME"] = "APERTURE"
+        aperture_hdu_average.header.set(
+            "NPIXSAP", None, "Number of pixels in optimal aperture"
+        )
+        aperture_hdu_average.header.set(
             "NPIXMISS", None, "Number of op. aperture pixels not collected"
         )
 
+        # Define extra FITS columns
+        cols = [
+            # Original FFI column of lower-left pixel in TPF.
+            fits.Column(
+                name="CORNER1",
+                format="I",
+                unit="pixel",
+                array=self.corner[:, 1],
+            ),
+            # Original FFI row of lower-left pixel in TPF.
+            fits.Column(
+                name="CORNER2",
+                format="I",
+                unit="pixel",
+                array=self.corner[:, 0],
+            ),
+            # 3D pixel quality mask
+            fits.Column(
+                name="PIXEL_QUALITY",
+                format=str(self.corr_flux[0].size) + "I",
+                dim=dims,
+                disp="B16.16",
+                array=self.pixel_quality,
+            ),
+            # Aperture as a function of time.
+            # Aperture has values 0 and 2, where 0/2 indicates the pixel is outside/inside the aperture.
+            # This format is used to be consistent with the aperture HDU from SPOC.
+            fits.Column(
+                name="APERTURE",
+                format=str(self.corr_flux[0].size) + "J",
+                dim=dims,
+                array=self.aperture_mask.astype("int32") * 2,
```

Comment:
since this one isn't read by `lightkurve`, then does it need to have the SPOC convention? if not, we  could keep it as boolean (?)
In the long run it could save some significant storage

Reply:
It's a good question. I used the SPOC convention so both the average aperture and aperture as a function of time would be consistent. Also, it still allows you to plot the aperture as a function of time with lightkurve. If we do a for loop through frames, we can access the aperture at that time and include it in tpf.plot(). If the aperture is represented by bit 2, lk will plot the aperture but otherwise it won't. I think it is still useful to have the option to plot the aperture with lk?

Reply:
ok, that makes sense.